automation/check-patch.e2e-kubemacpool-functests.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash -xe
+
+# This script should be able to execute KubeMacPool
+# functional tests against Kubernetes cluster with
+# CNAO built with latest changes, on any
+# environment with basic dependencies listed in
+# check-patch.packages installed and docker running.
+#
+# yum -y install automation/check-patch.packages
+# automation/check-patch.e2e-kubemacpool-functests.sh
+
+teardown() {
+    make cluster-down
+}
+
+main() {
+    export KUBEVIRT_PROVIDER='k8s-1.17'
+
+    source automation/check-patch.setup.sh
+
+    # Spin up Kubernetes cluster
+    cd ${TMP_PROJECT_PATH}
+    make cluster-down
+    make cluster-up
+    trap teardown EXIT SIGINT SIGTERM SIGSTOP
+
+    # Deploy CNAO with latest changes
+    make cluster-operator-push
+    make cluster-operator-install
+
+    # Deploy Kubevirt
+    export KUBEVIRT_VERSION=v0.29.0
```

Comment:
Maybe we should take latest released kubevirt since so we streamline everything, or at lastet latest from a specific major.minor so we get bugfixed release.

Reply:
If we use deploy-kubevirt.sh then we need to maintain this (for example bumping the kubevirt version) here and also on KMP repo.
I suggest to drop this script and instead run kubevirt deployment via the kubemacpool repo after cloning:
```
KUBEVIRT_PROVIDER=extarnal KUBECONFIG=$(_kubevirtci/cluster-up/kubeconfig.sh) make cluster-up
```

This way we don't need to maintain 2 KUBEVIRT_VERSION references and only need to bump via KMP repo

Reply:
Which script to drop?

Reply:
I agree we should keep this on latest to lower the maintenance cost. Also, we should be compatible with the latest KubeVirt, testing old one would not really help us with integration testing.

Reply:
I think we should use KMP's new function [getLatestPatchVersion](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/master/cluster/up.sh#L19)